src/platform/android/DnssdImpl.cpp

```diff
@@ -254,16 +255,38 @@ void HandleResolve(jstring instanceName, jstring serviceType, jstring address, j
     JniUtfString jniServiceType(env, serviceType);
     JniUtfString jniAddress(env, address);
     Inet::IPAddress ipAddress;
+    Inet::InterfaceId iface;
 
     VerifyOrReturn(strlen(jniInstanceName.c_str()) <= Operational::kInstanceNameMaxLength, dispatch(CHIP_ERROR_INVALID_ARGUMENT));
     VerifyOrReturn(strlen(jniServiceType.c_str()) <= kDnssdTypeAndProtocolMaxSize, dispatch(CHIP_ERROR_INVALID_ARGUMENT));
     VerifyOrReturn(CanCastTo<uint16_t>(port), dispatch(CHIP_ERROR_INVALID_ARGUMENT));
-    VerifyOrReturn(Inet::IPAddress::FromString(jniAddress.c_str(), ipAddress), dispatch(CHIP_ERROR_INVALID_ARGUMENT));
+
+    char * addrStr   = const_cast<char *>(jniAddress.c_str());
```

Comment:
Why can't we use the PeerAddress code here instead of copying it?

Reply:
So before your changes transport depends on access and access depends on platform/android?

Reply:
Yeah, `src/access` depends on `src/platform` (which is platform:android for us). Even removing that dep just for fun points out another cycle with `src/credentials`:

```
2022-06-10 15:31:52 INFO      //src/platform:platform ->
2022-06-10 15:31:52 INFO      //src/platform/android:android ->
2022-06-10 15:31:52 INFO      //src/transport:transport ->
2022-06-10 15:31:52 INFO      //src/credentials:credentials ->
2022-06-10 15:31:52 INFO      //src/platform:platform
```

Reply:
Hmm.  Why does transport depend on access then?

Can we put the parsing code someplace that is not transport?   e.g. have it return an IPAddress and interface and put it in inet?

Reply:
Done, added an overload to `IPAddress-StringFuncts` that will output the `IPAddress` and `InterfaceId` and shared this with `DnssdImpl` and `PeerAddress` both.